lib/fizzy/capi.cpp

```diff
@@ -113,23 +123,23 @@ inline fizzy::ExecutionResult unwrap(const FizzyExecutionResult& result) noexcep
 inline fizzy::ExecuteFunction unwrap(FizzyExternalFn c_function, void* c_host_context)
 {
     static constexpr fizzy::HostFunctionPtr function =
-        [](std::any& host_context, fizzy::Instance& instance, const fizzy::Value* args,
-            int depth) noexcept {
+        [](std::any& host_ctx, fizzy::Instance& instance, const fizzy::Value* args,
+            fizzy::ExecutionContext& ctx) noexcept {
             const auto [c_func, c_host_ctx] =
-                *std::any_cast<std::pair<FizzyExternalFn, void*>>(&host_context);
-            return unwrap(c_func(c_host_ctx, wrap(&instance), wrap(args), depth));
+                *std::any_cast<std::pair<FizzyExternalFn, void*>>(&host_ctx);
+            return unwrap(c_func(c_host_ctx, wrap(&instance), wrap(args), wrap(ctx)));
         };
 
     return {function, std::make_any<std::pair<FizzyExternalFn, void*>>(c_function, c_host_context)};
 }
 
 inline FizzyExternalFunction wrap(fizzy::ExternalFunction external_func)
 {
-    static constexpr FizzyExternalFn c_function = [](void* context, FizzyInstance* instance,
-                                                      const FizzyValue* args,
-                                                      int depth) -> FizzyExecutionResult {
-        auto* func = static_cast<fizzy::ExternalFunction*>(context);
-        return wrap((func->function)(*unwrap(instance), unwrap(args), depth));
+    static constexpr FizzyExternalFn c_function =
+        [](void* host_context, FizzyInstance* instance, const FizzyValue* args,
```

Comment:
Nit: it seems some places use `_ctx`, while this uses `_context`. I think a while back we argued to use the long form, but if we go for the short one, please make it consistent.

Reply:
Shadowing issues here.

Do you want to change everything to `context`?

Reply:
It was `_context` before, I'm happy with `_ctx` too.